src/cnaas_nms/api/auth.py

```diff
@@ -0,0 +1,85 @@
+from authlib.integrations.base_client.errors import MismatchingStateError
+from flask import current_app, redirect, url_for
+from flask_restx import Namespace, Resource
+from requests.models import PreparedRequest
+
+from cnaas_nms.api.generic import empty_result
+from cnaas_nms.app_settings import auth_settings
+from cnaas_nms.tools.log import get_logger
+from cnaas_nms.tools.security import get_identity, login_required
+from cnaas_nms.version import __api_version__
+
+logger = get_logger()
+api = Namespace("auth", description="API for handling auth", prefix="/api/{}".format(__api_version__))
+
+
+class LoginApi(Resource):
+    def get(self):
+        """Function to initiate a login of the user.
+        The user will be sent to the page to login.
+        Our client info will also be checked.
+
+        Note:
+            We also discussed adding state to this function.
+            That way you could be sent to the same page once you logged in.
+            We would put the relevant information in a dictionary,
+            base64 encode it and sent it around as a parameter.
+            For now the application is small and it didn't seem needed.
+
+        Returns:
+            A HTTP redirect response to OIDC_CONF_WELL_KNOWN_URL we have defined.
+            We give the auth call as a parameter to redirect after login is successfull.
+
+        """
+        if not auth_settings.OIDC_ENABLED:
+            return empty_result(status="error", data="Can't login when OIDC disabled"), 500
+        oauth_client = current_app.extensions["authlib.integrations.flask_client"]
+        redirect_uri = url_for("auth_auth_api", _external=True)
+
+        return oauth_client.connext.authorize_redirect(redirect_uri)
+
+
+class AuthApi(Resource):
+    def get(self):
+        """Function to authenticate the user.
+        This API call is called by the OAUTH login after the user has logged in.
+        We get the users token and redirect them to right page in the frontend.
+
+        Returns:
+            A HTTP redirect response to the url in the frontend that handles the repsonse after login.
+            The access token is a parameter in the url
+
+        """
+
+        oauth_client = current_app.extensions["authlib.integrations.flask_client"]
+
+        try:
+            token = oauth_client.connext.authorize_access_token()
```

Comment:
Maybe we can catch error of type "authlib.integrations.base_client.errors.OAuthError: unauthorized_client" here also, if the oauth server doesn't have the client_id registered the nms api just gives "Internal server error":
```
  File "/home/johanmarcusson/python/cnaas-nms/git/src/cnaas_nms/api/auth.py", line 57, in get
    token = oauth_client.connext.authorize_access_token()
  File "/home/johanmarcusson/.pyenv/versions/py373/lib/python3.7/site-packages/authlib/integrations/flask_client/apps.py", line 85, in authorize_access_token
    raise OAuthError(error=error, description=description)
authlib.integrations.base_client.errors.OAuthError: unauthorized_client: 
```

Reply:
Great idea, I implemented error handling with a hint that information is missing.